lib/clustering/ClusterRPC.ts

```
@@ -0,0 +1,469 @@
+ *         handler({object} payload, {string} uids, {function} callback)
+ *     handler callback must be called when worker is done with the command:
+ *         callback({Error|null} error, {any} [result])
+ * @return {undefined}
+ * }
+ */
+export function setupRPCWorker(handlers: HandlersMap) {
+    process.on('message', (message: RPCCommandMessage | RPCCommandResultsMessage) => {
+        if (_isRpcMessage(message)) {
+            _handleWorkerMessage(message, handlers);
+        }
+    });
+}
+
+/**
+ * Send a command for workers to execute in parallel, and wait for results
+ *
+ * @param {string} toWorkers - which workers should execute the command
+ *     Currently the only supported value is "*", meaning all workers will
+ *     execute the command
+ * @param {string} toHandler - name of handler that will execute the
+ * command in workers, as declared in setupRPCWorker() parameter object
+ * @param {string} uids - unique identifier of the command, must be
+ * unique across all commands in progress
+ * @param {object} payload - message payload, sent as-is to the handler
+ * @param {function} cb - command callback
+ * @returns {undefined}
+ */
+export function sendWorkerCommand(
+    toWorkers: string,
+    toHandler: string,
+    uids: string,
+    payload: object,
+    callback: CommandCallback
+) {
+    if (typeof uids !== 'string') {
+        rpcLogger.error('missing or invalid "uids" field', { uids });
+        return callback(errors.MissingParameter);
+    }
+    if (uidsToCallback[uids] !== undefined) {
+        rpcLogger.error('a command is already in progress with same uids', { uids });
+        return callback(errors.OperationAborted);
+    }
+    rpcLogger.info('sending command', { uids, payload });
+    uidsToCallback[uids] = callback;
+    const message: RPCCommandMessage = {
+        type: 'cluster-rpc:command',
+        toWorkers,
+        toHandler,
+        uids,
+        payload,
+    };
+    return process.send?.(message);
```

If `process.send` is undefined no worker will receive the call, so it might be worth failing early at init if the IPC channel is not available.